src/input/base-input.js

```diff
@@ -55,12 +58,39 @@ class BaseInput<T: EventTarget> extends React.Component<
   };
 
   componentDidMount() {
-    const {autoFocus, inputRef} = this.props;
-    if (autoFocus && inputRef.current) {
-      inputRef.current.focus();
+    const {inputRef, autoFocus, clearable} = this.props;
+    if (inputRef.current) {
+      if (autoFocus) {
+        inputRef.current.focus();
+      }
+      if (clearable) {
+        inputRef.current.addEventListener('keyup', this.onInputKeyUp);
+      }
     }
   }
 
+  componentWillUnmount() {
+    const {inputRef, clearable} = this.props;
+    if (clearable && inputRef.current) {
+      inputRef.current.removeEventListener('keyup', this.onInputKeyUp);
+    }
+  }
+
+  onInputKeyUp = (e: SyntheticKeyboardEvent<T>) => {
```

Comment:
doing this on `keyDown` might be better so it's same as the native search input?

Reply:
oh good point - i assumed it was keyup